src/content/docs/en/basics/rendering.mdx

```diff
@@ -0,0 +1,75 @@
+
+By default, all Astro projects are configured to be pre-rendered at build time (statically generated) to provide the most lightweight browser experience. The browser does not need to wait for any HTML to build because the server does not need to generate any pages on demand. Your site is not dependent on the performance of a backend data source, and once built, will remain available to visitors as a static site as long as your server is functioning.
+
+Prerendered pages in Astro can include:
+
+- [Client islands](/en/concepts/islands/) for interactive UI components in an otherwise static, pre-rendered page.
+- [Server islands]() for rendering dynamic or personalized content within a static page shell.
+- [View transitions](/en/guides/view-transitions/) for client-side routing with animation and state persistence across page navigation.
+- [Middleware](/en/guides/middleware/) (with some limitations) to intercept and transform response data from a request.
+
+:::tip
+Astro's default `static` mode is a powerful, modern-feeling choice for content-focused sites with the ability to add individual on-demand routes for dynamic or personalized pages when needed.
+:::
+
+### On-demand rendered routes
+
+<p>
+
+**`'static'`:** with `export const prerender = false` and an adapter<br />
+**`'server'`:** by default, with an adapter
+</p>
+  
+Since on-demand routes are generated per visit, they can be customized for each viewer. For example, a page rendered on demand can show a logged-in user their account information or display freshly updated data without requiring a full-site rebuild. On-demand rendering on the server at request time is also known as **server-side rendering (SSR)**.
+
+<ReadMore>Read more about [features available with on-demand rendering and an adapter](/en/guides/on-demand-rendering/#on-demand-rendering-features), such as cookies, Response and Request objects, HTML streaming, and more.</ReadMore>
+
+You may want to render a route on demand if you need the following:
+
+- **API endpoints**: Create specific pages that function as API endpoints for tasks like database access, authentication, and authorization while keeping sensitive data hidden from the client.
+
+- **Protected pages**: Restrict access to a page based on user privileges, by handling user access on the server.
+
+- **Frequently changing content**: Generate individual pages without requiring a static rebuild of your site. This is useful when the content of a page updates frequently, for example displaying data from an API called dynamically with `fetch()`.
+
+:::tip
+On-demand server rendering in Astro, especially when combined with [view transitions](/en/guides/view-transitions/), provides a true app experience without the JavaScript overhead of a client-side, single-page application. Even highly dynamic, interactive apps are possible.
```

Comment:
I am not sure if having both on-demand and server rendering should be the case. I guess we're doing this way so it's easier for people to say "oh, it's like SSR" -- but my feeling generally goes towards choosing one and keeping it to make things clear and straightforward, but sure it would make sense to mention at the start "On-demand rendering, similar to server-side rendering (SSR), yada yada yada"

Reply:
Yeah, as the most basic of content, I'm OK having this be a tiny bit more explicit/all-encompassing like this. I wouldn't do it often, but on the one page we're introducing our concepts, I think it's fine!